### PR TITLE
feat: dry run on reviewpad.yml change

### LINKED ISSUE #10: Run reviewpad on dry-run when reviewpad.yml is on patch
**Is your feature request related to a problem? Please describe.**
Currently, the reviewpad action fetches the `reviewpad.yml` configuration (to run) from the pull request `base`.

For instance, when creating a pull request from `B (head) -> A (base)`, the `reviewpad.yml` configuration is fetched from the branch `A`.

This approach makes it very hard to test a new `reviewpad.yml` configuration.

**Describe the solution you'd like**
A user should be able to update the `reviewpad.yml` on a pull request and see the output of the updated `reviewpad.yml` on the created pull request.

This means that the reviewpad action should fetch `reviewpad.yml` configuration from `head`, instead of `base`.

However, for security reasons, in this scenario reviewpad should run on `dry-run`. This means, that reviewpad should create a report with the dry-run output and add it as a comment to the pull request.

**Describe alternatives you've considered**
_none_

**Additional context**
The decision of always fetching `reviewpad.yml` from `base` comes from security reasons.

Once reviewpad actions runs with a github action token (ultimately with a user token), actions such `merge` could be taken without any restrictions.

### PR DESCRIPTION
**Pull request by marcelosousa**
## Description

This pull request introduces the following changes in the behaviour of the action:
- If the reviewpad.yml file has been changed by a pull request, it will run Reviewpad in dry run mode using the reviewpad.yml file from the head branch of the pull request.

This pull request subsumes https://github.com/reviewpad/action/pull/12 and is co-authored by @zolamk !

## Related issue

Closes #10 

## Type of change

Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How was this tested?

This feature requires extensive functional testing.

## Checklist

- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works 
- [x] I have ran `task check -f` and have no issues

### CODE REVIEW

**Comment by marcelosousa** [comment]
@zolamk FYI: you should have push access to this branch in case we need to do more improvements.

**Comment by marcelosousa** [comment]
@zolamk: FYI, pushes to branches starting with `external` generate an image at https://hub.docker.com/repository/docker/reviewpad/action-dev with the tag `external_branch-name`. 

This means that there is an image already for this branch: https://hub.docker.com/layers/action-dev/reviewpad/action-dev/dev-external_dry-run-reviewpad/images/sha256-2851af01cae4bc1af5f4921de606241105c07d281eb18b94e28a786a0f6893c0?context=repo.

This image is used for functional test.

File: utils/file.go [hunk 1]
```diff
@@ -43,21 +45,29 @@ func LoadReviewpadFile(ctx context.Context, filePath string, client *github.Clie
...
-		rawBaseFile, err := downloadReviewPadFile(ctx, filePath, client, pullRequest.Base)
+		rawBaseFile, err := downloadFileFromHost(ctx, filePath, client, pullRequest.Base)
 		if err != nil {
 			if strings.HasPrefix(err.Error(), "no file named") {
 				return true, nil
 			}
 			return false, err
 		}
 
+		// TODO: check if this Equal uses the hashes of the files.
```
Comment (zolamk) [inline utils/file.go:70]: I think [bytes.Equal](https://cs.opensource.google/go/go/+/refs/tags/go1.18.4:src/bytes/bytes.go) just casts the passed byte slices to string and compares the casted values

**Review by zolamk — APPROVED** [review]

**Review by ferreiratiago — APPROVED** [review]
LGTM

**Pull request merged**